### PR TITLE
Add change-password form

### LINKED ISSUE #333: Allow users to change their password
Is there an endpoint? Putting `{users: {password: 'my new password'}}` returns a 422.

### PR DESCRIPTION
**Pull request by brian-c**
Allows users to change their passwords from the settings page

Closes #333.

### CODE REVIEW

File: app/pages/settings.cjsx [hunk 1]
```diff
@@ -9,6 +9,90 @@ apiClient = require '../api/client'
...
+          </tr>
+        </tbody>
+      </table>
+
+      <p>
+        <button type="submit" className="standard-button" disabled={not @state.old or not @state.new or @tooShort() or @doesntMatch() or @state.inProgress}>Change</button>{' '}
+
+        {if @state.inProgress
+          <i className="fa fa-spinner fa-spin form-help"></i>
+        else if @state.success
```
Comment (chrissnyder) [inline app/pages/settings.cjsx]: Nitpicky, but `@state.success` is never used, due to having to signOut/signIn to get a new session.

**Comment by chrissnyder** [comment]
Other than the above comment, looks good to me.

File: app/pages/settings.cjsx (hunk 1 above)
Comment (brian-c) [inline app/pages/settings.cjsx]: I'd actually like to leave this. The real issue is that the auth modules needs to sign out and back in. There might be a way around that, but I don't think it's worth investing too much time in right now.

Comment (chrissnyder) [inline app/pages/settings.cjsx]: Roger that.

**Pull request merged**